lib/beaker/host.rb

```diff
@@ -92,11 +92,32 @@ def port_open? port
           TCPSocket.new(reachable_name, port).close
           return true
         end
-      rescue Errno::ECONNREFUSED, Timeout::Error, Errno::ETIMEDOUT
+      rescue Errno::ECONNREFUSED, Timeout::Error, Errno::ETIMEDOUT, Errno::EHOSTUNREACH
         return false
       end
     end
 
+    # Wait for a port on the host.  Useful for those occasions when you've called
+    # host.reboot and want to avoid spam from subsequent SSH connections retrying
+    # to connect from say retry_on()
+    def wait_for_port(port, timeout=60, interval=1)
```

Comment:
@spjmurray thanks for the PR! For methods that are waiting, we would like to see the `repeat_fibonacci_style_for` method used to exponentially backoff retry attempts. You can see the definition of the method [here](https://github.com/puppetlabs/beaker/blob/a6c2297b7dc5959a55f1db538a967f15bfe9ab2a/lib/beaker/shared/repetition.rb#L14) and an example of it being used [here](https://github.com/puppetlabs/beaker/blob/a6c2297b7dc5959a55f1db538a967f15bfe9ab2a/lib/beaker/host/windows/exec.rb#L59). Instead of a seconds based timeout, I think that a number of retry attempts should be the parameter configurable here.

Also, since this is a new method, we would to see some spec coverage; please take a look at the contributing doc, specifically [here](https://github.com/puppetlabs/beaker/blob/master/CONTRIBUTING.md#testing) on what we are looking for.

Please respond in this PR with any questions you have. Thanks!